app/views/course/assessment/assessments/index.json.jbuilder

```diff
@@ -1,28 +1,130 @@
+
+  assessment_with_loaded_timeline = @items_hash[assessment.id].actable
+  can_attempt_assessment = can?(:attempt, assessment_with_loaded_timeline)
+  json.canAttempt can_attempt_assessment
+
+  submissions = submissions_hash[assessment.id]
+  attempting_submission = submissions.find(&:attempting?)
+  submitted_submission = submissions.find { |submission| !submission.attempting? }
+
+  action_url = nil
+  if !current_course_user || !can_attempt_assessment
+    status = 'unavailable'
+  elsif cannot?(:access, assessment) && can_attempt_assessment
+    status = 'locked'
+    action_url = course_assessment_path(current_course, assessment)
+  elsif !attempting_submission.nil?
+    status = 'attempting'
+    action_url = edit_course_assessment_submission_path(current_course, assessment, attempting_submission)
+  elsif !submitted_submission.nil?
+    status = 'submitted'
+    action_url = edit_course_assessment_submission_path(current_course, assessment, submitted_submission)
+  else
+    status = 'open'
+    action_url = course_assessment_attempt_path(current_course, assessment)
+  end
+
+  json.status status
+  json.actionUrl action_url
+
+  can_view_submissions = can?(:view_all_submissions, assessment)
+  json.submissionsUrl course_assessment_submissions_path(current_course, assessment) if can_view_submissions
+
+  can_edit_assessment = can?(:manage, assessment)
+  json.editUrl edit_course_assessment_path(current_course, assessment) if can_edit_assessment
+
+  if achievements_enabled
+    achievement_conditionals = @conditional_service.achievement_conditional_for(assessment)
+
+    top_dependants = achievement_conditionals.first(3)
+    json.topDependants top_dependants do |achievement|
+      json.url course_achievement_path(current_course, achievement)
+      json.badgeUrl achievement_badge_path(achievement)
+      json.title achievement.title
+    end
+
+    dependants_count = achievement_conditionals.size
+    json.remainingDependantsCount dependants_count - top_dependants.size if dependants_count > top_dependants.size
+  end
+
+  json.baseExp assessment.base_exp if current_course.gamified? && assessment.base_exp > 0
+
+  assessment_time = @items_hash[assessment.id].time_for(current_course_user)
+  json.conditionSatisfied !condition_not_satisfied(
+    can_attempt_assessment,
+    assessment_with_loaded_timeline,
+    assessment_time
+  )
+
+  json.isStartTimeBegin !assessment_not_started(assessment_time)
+  json.startAt do
+    json.partial! 'course/lesson_plan/items/personal_or_ref_time', locals: {
+      item: @items_hash[assessment.id],
+      course_user: current_course_user,
+      attribute: :start_at,
+      datetime_format: :short
+    }
+  end
+
+  has_bonus_attributes = assessment_time.bonus_end_at.present? && assessment.time_bonus_exp > 0
+  if show_bonus_attributes? && has_bonus_attributes
+    json.timeBonusExp assessment.time_bonus_exp
+    json.isBonusEnded assessment.bonus_end_at <= Time.zone.now
```

Comment:
Good point. I didn't want to compare time on the front-end because (a) we should be always using server time for reconciliations, and (b) what if the user's computer has incorrect time? Then JavaScript's `new Date().now` might return the computer's time and compare to it instead (I did not test it, but I'd imagine so).